### PR TITLE
Fix grey gap in split omnibar during snap animation

### PR DESCRIPTION
**Pull request by aibrahim-**
Task/Issue URL: https://app.asana.com/1/137249556945/project/715106103902962/task/1212150815656027?focus=true

### Description
n split omnibar mode, releasing a scroll mid-collapse causes a grey gap where the navigation bar was. The AppBarLayout snap animation auto-collapses the omnibar, but          `CoordinatorLayoutHelper` stops receiving `onOverScrolled` callbacks during snap, leaving a stale `bottomMargin` on the browser layout.

Fix: Replace `CoordinatorLayoutHelper`'s async margin correction (via post()) with a synchronous correction in `TopOmnibarBrowserContainerLayoutBehavior.onDependentViewChanged`. This  fires on every frame during snap, to compute the correct margin. 

This will also fix another visual bug, where while scrolling down, there was a visible visual artifact which was happening because bottomMargin adjustments were taking place at later frames after the scrolling actually takes place.
 
### Steps to test this PR
- open DDG in split omnibar mode.
- slowly scroll down any site

### UI changes
| Before  | After |
|---------| ----- |
before|after

---

> [!NOTE]
> **Medium Risk**
> Changes CoordinatorLayout margin calculations during AppBarLayout scroll/snap, which can affect web content layout across devices and omnibar configurations. Risk is limited to UI behavior (no security/data logic) but could introduce new visual regressions if the margin math is off.
> 
> **Overview**
> Fixes the split/top omnibar “grey gap” by moving browser-container bottom-margin correction from an async `post` in `DuckDuckGoWebView.onOverScrolled` to a synchronous, per-frame adjustment in `TopOmnibarBrowserContainerLayoutBehavior.onDependentViewChanged`.
> 
> Adds `correctBottomMargin` logic (with a small 7dp offset during partial collapse to reduce flicker) and introduces an instrumentation test suite covering expanded/collapsed/partial and edge-case margin calculations.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 846404afa761a3c5d8da3ba0cb19f04d9b979198. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by aibrahim-** [comment]
* **#7883** <a href="https://app.graphite.com/github/pr/duckduckgo/Android/7883?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/duckduckgo/Android/7883?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: app/src/main/java/com/duckduckgo/app/browser/webview/BrowserContainerLayoutBehavior.kt [hunk 1]
```diff
@@ -58,8 +58,27 @@ class TopOmnibarBrowserContainerLayoutBehavior(
...
+        if (lp.bottomMargin == 0) return
+
+        val childBottom = child.bottom + child.translationY.toInt()
+        val parentBottom = parent.height
+        val diff = childBottom - parentBottom
+        val newMargin = if (diff > 0) diff else 0
+        if (lp.bottomMargin != newMargin) {
+            lp.bottomMargin = newMargin
+        }
+    }
```
[the remark this answers is not in the document]
Comment (aibrahim-) [reply app/src/main/java/com/duckduckgo/app/browser/webview/BrowserContainerLayoutBehavior.kt:84]: we intentionally avoid it. Setting lp.bottomMargin directly (without child.layoutParams = lp) mutates the existing params without triggering requestLayout(). That's the correct 
  approach here because we're already inside onDependentViewChanged, which runs during a layout pass. Triggering requestLayout() from within layout causes recursive layout and ANRs —  
  which we hit earlier in this session.

 The margin takes effect on the next layout pass, which happens naturally since the AppBarLayout snap animation keeps driving layout changes.

**Comment by aibrahim-** [comment]
This is a great catch, thanks for testing it! 

I reverted back the helper as it was doing the initial bottomMargin calculations correctly, my changes will only take effect in a scrollable page. 

You can find latest build in the asana task.
